pkg/policy/groups/helpers.go

```diff
@@ -123,24 +169,74 @@ func updateDerivativeStatus(cnp *cilium_v2.CiliumNetworkPolicy, derivativeName s
+	}
+
+	return updateDerivativeCCNPStatus(cnp, status, derivativeName)
+}
+
+func updateDerivativeCNPStatus(cnp *cilium_v2.CiliumNetworkPolicy, status cilium_v2.CiliumNetworkPolicyNodeStatus,
+	derivativeName string) error {
 	// This CNP can be modified by cilium agent or operator. To be able to push
 	// the status correctly fetch the last version to avoid updates issues.
-	k8sCNPStatus, clientErr := k8s.CiliumClient().CiliumV2().
-		CiliumNetworkPolicies(cnp.ObjectMeta.Namespace).
+	k8sCNP, clientErr := k8s.CiliumClient().CiliumV2().CiliumNetworkPolicies(cnp.ObjectMeta.Namespace).
 		Get(context.TODO(), cnp.ObjectMeta.Name, v1.GetOptions{})
+
 	if clientErr != nil {
-		return fmt.Errorf("Cannot get Kubernetes policy: %s", clientErr)
+		return fmt.Errorf("cannot get Kubernetes policy: %s", clientErr)
 	}
-	if k8sCNPStatus.ObjectMeta.UID != cnp.ObjectMeta.UID {
+
+	if k8sCNP.ObjectMeta.UID != cnp.ObjectMeta.UID {
 		// This case should not happen, but if the UID does not match make sure
 		// that the new policy is not in the cache to not loop over it. The
 		// kubernetes watcher should take care about that.
-		groupsCNPCache.DeleteCNP(k8sCNPStatus)
-		return fmt.Errorf("Policy UID mistmatch")
+		groupsCNPCache.DeleteCNP(k8sCNP)
+		return fmt.Errorf("policy UID mistmatch")
 	}
-	k8sCNPStatus.SetDerivedPolicyStatus(derivativeName, status)
-	groupsCNPCache.UpdateCNP(k8sCNPStatus)
-	// TODO: switch to JSON Patch
-	_, err = k8s.CiliumClient().CiliumV2().CiliumNetworkPolicies(cnp.ObjectMeta.Namespace).UpdateStatus(context.TODO(), cnp, v1.UpdateOptions{})
+
+	k8sCNP.SetDerivedPolicyStatus(derivativeName, status)
+	groupsCNPCache.UpdateCNP(k8sCNP)
+
+	// TODO: Switch to JSON patch.
+	_, err := k8s.CiliumClient().CiliumV2().CiliumNetworkPolicies(cnp.ObjectMeta.Namespace).
+		UpdateStatus(context.TODO(), k8sCNP, v1.UpdateOptions{})
+
+	return err
+}
+
+func updateDerivativeCCNPStatus(cnp *cilium_v2.CiliumNetworkPolicy, status cilium_v2.CiliumNetworkPolicyNodeStatus,
+	derivativeName string) error {
+	k8sCCNP, clientErr := k8s.CiliumClient().CiliumV2().CiliumClusterwideNetworkPolicies().
+		Get(context.TODO(), cnp.ObjectMeta.Name, v1.GetOptions{})
+
+	if clientErr != nil {
+		return fmt.Errorf("cannot get Kubernetes policy: %s", clientErr)
+	}
+
+	if k8sCCNP.ObjectMeta.UID != cnp.ObjectMeta.UID {
+		// This case should not happen, but if the UID does not match make sure
+		// that the new policy is not in the cache to not loop over it. The
+		// kubernetes watcher should take care about that.
+		groupsCNPCache.DeleteCNP(&cilium_v2.CiliumNetworkPolicy{
+			ObjectMeta: k8sCCNP.ObjectMeta,
+		})
+		return fmt.Errorf("policy UID mistmatch")
+	}
+
+	k8sCCNP.SetDerivedPolicyStatus(derivativeName, status)
+	groupsCNPCache.UpdateCNP(&cilium_v2.CiliumNetworkPolicy{
+		TypeMeta:   k8sCCNP.TypeMeta,
+		ObjectMeta: k8sCCNP.ObjectMeta,
+		Spec:       k8sCCNP.Spec,
+		Specs:      k8sCCNP.Specs,
+		Status:     k8sCCNP.Status,
+	})
+
+	// TODO: Switch to JSON patch
```

Comment:
Is this meant for this PR or as a followup? Might be nice to provide context why.

Reply:
This will be a follow-up. This is an old TODO actually that we had from earlier, so I wanted to take this up in a separate PR.